pennylane/ops/op_math/controlled_ops.py

```diff
@@ -117,68 +123,75 @@ class ControlledQubitUnitary(ControlledOp):
 
     @classmethod
     def _unflatten(cls, data, metadata):
-        return cls(
-            data[0], control_wires=metadata[0], control_values=metadata[1], work_wires=metadata[2]
-        )
+        return cls(data[0], metadata[0], control_values=metadata[1], work_wires=metadata[2])
 
     # pylint: disable=arguments-differ, too-many-arguments, unused-argument, too-many-positional-arguments
     @classmethod
     def _primitive_bind_call(
         cls,
         base,
-        control_wires: WiresLike,
-        wires: WiresLike = (),
+        wires: WiresLike,
+        control_wires: WiresLike = "unset",
         control_values=None,
         unitary_check=False,
         work_wires: WiresLike = (),
     ):
-        wires = Wires(() if wires is None else wires)
+        cls._deprecate_control_wires(control_wires)
+
         work_wires = Wires(() if work_wires is None else work_wires)
 
         if hasattr(base, "wires") and len(wires) != 0:
             warnings.warn(
                 "base operator already has wires; values specified through wires kwarg will be ignored."
             )
-            wires = Wires(())
+            # wires = Wires(())
 
-        all_wires = control_wires + wires
         return cls._primitive.bind(
-            base, control_wires=all_wires, control_values=control_values, work_wires=work_wires
+            base, wires=wires, control_values=control_values, work_wires=work_wires
         )
 
+    @classmethod
```

Comment:
Making it a class method feels strange, can you just move it out of the class and making it a regular function instead?

Reply:
Does it even need to be abstracted at all?

Reply:
I abstracted it because there could be several different places to call the deprecation warnings. Right now besides `__init__` we also need to keep `_primitive_bind_call` same interface as `__init__` so another warning put there. Probably after review we will need more the I missed out